app/vmagent/remotewrite/remotewrite.go

```diff
@@ -235,8 +235,8 @@ func Stop() {
 // Push sends wr to remote storage systems set via `-remoteWrite.url`.
 //
 // Note that wr may be modified by Push due to relabeling and rounding.
-func Push(wr *prompbmarshal.WriteRequest) {
-	PushWithAuthToken(nil, wr)
+func Push(at *auth.Token, wr *prompbmarshal.WriteRequest) {
+	PushWithAuthToken(at, wr)
```

Comment:
I'd inline the `PushWithAuthToken()` function here in order to eliminate an additional function call, and then substitute the `PushWithAuthToken()` with `Push()` function everywhere.